### PR TITLE
Products Page & Product Detials Page

### PR DESCRIPTION
**Pull request by DonWick32**
- Created Products Page
- Created Product Details Page

### CODE REVIEW

File: lib/views/home.dart [hunk 1]
```diff
@@ -1,10 +1,189 @@
+// ignore_for_file: prefer_const_constructors
```
Comment (Ram-20062003) [inline lib/views/home.dart]: Do not use comments to override the errors . It is not good style of coding. Remove comments and resolve the issue

File: lib/views/home.dart [hunk 2]
```diff
@@ -1,10 +1,189 @@
...
+
 import 'package:flutter/material.dart';
 import 'package:get/get_state_manager/get_state_manager.dart';
 import 'package:thirumathikart_app/controllers/home_controller.dart';
+import 'package:thirumathikart_app/config/themes.dart';
+// ignore: import_of_legacy_library_into_null_safe
```
Comment (Ram-20062003) [inline lib/views/home.dart]: Same here remove commented lines( the library version is not supporting null safety ,change version)

File: lib/views/product_detail.dart [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+import 'package:flutter/material.dart';
+import 'package:thirumathikart_app/models/product.dart';
+import 'package:thirumathikart_app/config/themes.dart';
+
+class Product_Detail extends StatefulWidget {
+  late Product product;
+  Product_Detail(this.product);
+
+  @override
+  // ignore: library_private_types_in_public_api
```
Comment (Ram-20062003) [inline lib/views/product_detail.dart]: Do not use a stateful widget. instead use controllers. Refer getx package.

File: lib/views/products.dart [hunk 4]
```diff
@@ -0,0 +1,252 @@
+// ignore_for_file: must_be_immutable, non_constant_identifier_names, use_key_in_widget_constructors, library_private_types_in_public_api, no_logic_in_create_state
```
Comment (Ram-20062003) [inline lib/views/products.dart]: same here remove this

File: lib/views/product_detail.dart [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+            ),
+            Text(
+              '${product.name!} (1 ${product.unit!})',
+              style: const TextStyle(
+                fontSize: 30,
+                fontWeight: FontWeight.bold,
+              ),
+            ),
+            Text(
+              '₹ ${product.price}',
```
Comment (Ram-20062003) [inline lib/views/product_detail.dart]: Make sure product price changes as it increment and decrement

**Review by Ram-20062003 — CHANGES_REQUESTED** [review]

File: lib/views/products.dart [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+                    ],
+                  ),
+                ),
+              ),
+              SizedBox(
+                height: MediaQuery.of(context).size.height,
+                child: ListView.builder(
+                  itemBuilder: (ctx, index) => GestureDetector(
+                    onTap: () {
+                      Navigator.push(
```
Comment (Ram-20062003) [inline lib/views/products.dart]: Use getX here

File: lib/views/products.dart [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+  }
+
+  @override
+  Widget build(BuildContext context) => Scaffold(
+        appBar: AppBar(
+          centerTitle: true,
+          title: customSearchBar,
+          leading: GestureDetector(
+            onTap: () {
+              Navigator.pop(context);
```
Comment (Ram-20062003) [inline lib/views/products.dart]: Use getX here

File: lib/views/product_detail.dart [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+          title: Text(
+            product.name!,
+            style: const TextStyle(
+              fontSize: 20,
+              fontWeight: FontWeight.bold,
+            ),
+          ),
+          leading: GestureDetector(
+            onTap: () {
+              Navigator.pop(context);
```
Comment (Ram-20062003) [inline lib/views/product_detail.dart]: Use getx here

File: lib/controllers/products_controller.dart [hunk 9]
```diff
@@ -0,0 +1,35 @@
+import 'package:flutter/material.dart';
+import 'package:get/get.dart';
+import 'package:thirumathikart_app/models/product.dart';
+
+class ProductsController extends GetxController {
+  final textController = TextEditingController();
+  final items = [].obs;
+  final c = 0.obs;
```
Comment (Ram-20062003) [inline lib/controllers/products_controller.dart]: give proper namings here

File: lib/controllers/products_controller.dart [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+    items(productsList);
+  }
+
+  int length() => items.length;
+
+  void set(int i, bool a) {
+    isSelected[i] = a;
+  }
+
+  void done() {
```
Comment (Ram-20062003) [inline lib/controllers/products_controller.dart]: here too

File: lib/views/product_detail.dart [hunk 11]
```diff
@@ -0,0 +1,245 @@
...
+
+class ProductDetail extends GetView<ProductDetailsController> {
+  final Product product;
+  final List<Product> productList;
+  const ProductDetail(this.product, this.productList, {Key? key})
+      : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final productDetailsController = Get.put(ProductDetailsController());
```
Comment (Ram-20062003) [inline lib/views/product_detail.dart]: Do not initalize controller here . create your own bindings and do it. Refer other pages for it.

File: lib/views/products.dart [hunk 12]
```diff
@@ -0,0 +1,234 @@
...
+import 'package:thirumathikart_app/config/themes.dart';
+import 'package:thirumathikart_app/views/home.dart';
+import 'package:thirumathikart_app/views/product_detail.dart';
+import 'package:get/get.dart';
+
+class Products extends GetView<ProductsController> {
+  final List<Product> productsListMain;
+  Products(this.productsListMain, {Key? key}) : super(key: key);
+  final FocusNode focusNode = FocusNode();
+  final productsController = Get.put(ProductsController());
```
Comment (Ram-20062003) [inline lib/views/products.dart]: Create a binding for it and initialize controller over there

File: lib/views/products.dart [hunk 13]
```diff
@@ -0,0 +1,234 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    if (productsController.c.value == 0) {
+      productsController.copy(productsListMain);
+      productsController.done();
+    }
+    return Scaffold(
+      appBar: AppBar(
```
Comment (Ram-20062003) [inline lib/views/products.dart]: use the existing app bar

**Review by Ram-20062003 — CHANGES_REQUESTED** [review]

**Review by Ram-20062003 — APPROVED** [review]

**Review by Muhesh7 — APPROVED** [review]

**Pull request merged**